libcsm/tests/test_os.py

```diff
@@ -55,18 +57,48 @@ def test_run_command(self) -> None:
             assert isinstance(result.return_code, int)
             assert result.stdout or result.stderr
 
-    def test_run_command_unicode_error(self) -> None:
+    def test_run_command_decode(self) -> None:
         """
-        Assert that UnicodeDecodeErrors are caught by the run_command 
-        and a non-zero return code is recieved.
+        Assert that a ``_CLI`` object successfully decodes its ``stdout`` and
+        ``stderr``.
+        """
+        command = ['ls', '-l']
+        shell_command = 'ls -l'
+        bad_command = 'foo!!!'
+        no_shell_result = run_command(shell_command)
+        shell_result = run_command(
+            shell_command,
+            in_shell=True,
+            charset='utf-8'
+        )
+        bad_result = run_command(bad_command)
+        command_result = run_command(command)
+        for result in [no_shell_result, shell_result, bad_result,
+                       command_result]:
+            assert result.duration > 0.0
+            assert isinstance(result.return_code, int)
+            assert result.stdout or result.stderr
+
+    def test_run_command_decode_unicode_error(self) -> None:
+        """
+        Assert that ``UnicodeDecodeErrors`` are caught by the ``run_command``
+        and a non-zero return code is received.
         """
         # test with an invalid utf-8 byte
         byte_string = b'\x9c'
```

Comment:
Can we note this is a cp1252 oe char just to be clear what we're testing against?